### ISSUE TITLE
new normal and tangential velocity components parameter for Code table 4.2, Discipline 10

### ISSUE THREAD

**Issue by SibylleK** [wmo-im/GRIB2#253]
### Initial request

DWD requests two new oceanographic parameters for Code table 4.2.  

In case of meteorological products, the following two entries do exist in Code table 4.2:  
Product discipline 0 - Meteorological products, parameter category 2: momentum  
|Number|Parameter|Units|
|:-----|:--------|:----|
|34 | Normal wind component (see Note 2)  | m s-1 |
|35 | Tangential wind component (see Note 2)  | m s-1 |

Note 2: In relation to local coordinate axes at a cell edge. 

We require corresponding parameters for ocean modelling.

### Amendment details

Add in Code table 4.2  

Product discipline 10 - Oceanographic products, parameter category 4: subsurface properties  
|Number|Parameter|Units|
|:-----|:--------|:----|
|53 | Normal component of water velocity (see Note 3) | m s-1 |
|54 | Tangential component of water velocity (see Note 3) | m s-1 |

Notes:  
...  
(3) In relation to local coordinate axes at a (grid) cell edge.

### Requestor(s)

Sebastian Borchert (DWD)
Sibylle Krebber (DWD)

### Stakeholder(s)

Enter list of stakeholder(s).

### Publication(s)

Manual on Codes (WMO-No. 306), Volume I.2, GRIB code table 4.2, discipline 10, category 4

### Expected impact of change

None

**Comment by sebvi** [comment]
@SibylleK : I am struggling to understand how these 2 parameters differs from existing entries 23 to 26. 

Entries 23/24 are the water velocity components following the geographical axis (North/South and East/West) while the entries 25/26 are the water velocity components following the coordinates system (see the note attached to the entries).

**Comment by SibylleK** [comment]
@sebvi, I got some information from my colleague that says 
you are right, there are parallels between existing entries 25 & 26 (x- and y-components)
and requested new entries 53 & 54 (normal and tangential components).
However, we think there are differences, which justify new entries:

1. 1st: x- and y-components are with respect to a coordinate system, whose origin can be defined at arbitrary points.

      Normal and tangential components, however, imply that they are "normal" and "tangential" with respect to a specific geometric structure. In this case, the structure is a grid cell edge (or rather a grid cell face).

2nd: An x-y-coordinate system is typically associated with a global coordinate system, which is the same for all geometric points of interest 
       (e.g. the x-y-coordinate system defined on a meteorological f-plane tangential to the surface of the Earth at some location).
       Normal and tangential components, however, refer to a local coordinate system defined at the cell edge. 
       It changes from cell edge to cell edge.

3rd: x- and y-components typically imply a right-handedness of the coordinate system (x, y, z in this order).
      Normal and tangential components, however, leave open the handedness to the needs of the respective (ocean) model.
      (normal, tangential, z - in this order - may be right-handed or left-handed, as necessary).

**Comment by amilan17** [comment]
https://github.com/wmo-im/CCT/wiki/Teleconference.16.May.2024 notes: 
Sibylle introduced the issue.

**Comment by amilan17** [comment]
https://github.com/wmo-im/CCT/wiki/Teleconference.2024.June.4and5 notes:
@david-i-berry Can you review this proposal  (and see comments)?

**Comment by amilan17** [comment]
https://github.com/wmo-im/CCT/wiki/Teleconference.24and26.June.2024 notes:
still under discussion

**Comment by david-i-berry** [comment]
If we take the edge between cells B1 and B2 below, the normal will have a different sign depending on whether the edge is assigned to cell B1 or B2. How is this managed?

| | 1 |2 | 3
|-|-|-|-
|A | A1 | A2 | A3   
|B | B1 | B2 | B3    
|C | C1 | C2 | C3

**Comment by SibylleK** [comment]
@david-i-berry , I receive following answer from my colleague:

Sorry, referring to "grid **cell** edges" in our proposal was probably misunderstandable.
We should just have referred to "grid edges".  

You are right, if we consider grid cells with their (outward pointing) surface normals, 
The normal of B1 at the interface between B1 and B2 would be antiparallel to the normal of B2 there.  

The normal and tangential velocity components of our proposal, however, do not refer to the grid cells, 
but they just refer to the grid edges as such.  

Which of the two possible normal directions of an edge is chosen, 
and if the normal and tangential directions - in this order - form a right- or left-handed coordinate system,
should not be part of the parameter definition. 
It is a matter of individual definition and should be left to the (technical) conventions and needs of the particular ocean model.  

In summary, the parameters of this proposal are really just a one-to-one analogues 
of the already existing normal and tangential wind component parameters for atmospheric models (see above).

**Comment by amilan17** [comment]
https://github.com/wmo-im/CCT/wiki/Teleconference.2.July.2024 notes: 
team agrees with this proposal

**Comment by amilan17** [comment]
@SibylleK I updated the branch, but I was wondering if the numbers should be 52 and 53...
